### PR TITLE
feat: delete suggestion from record on search engine

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

Delete a suggestion from the record document on search engine when the suggestion is deleted using it's deletion endpoint.

Ref #4230 

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Running tests locally.
- [x] Checking manually that deleting a suggestion is affecting filters later on the UI.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/server/contexts/datasets.py [hunk 1]
```diff
@@ -1092,20 +1092,32 @@ async def upsert_suggestion(
 
 async def delete_suggestions(db: "AsyncSession", record: Record, suggestions_ids: List[UUID]) -> None:
```
Comment (frascuchon) [inline src/argilla/server/contexts/datasets.py]: this code may not be affected?

File: src/argilla/server/contexts/datasets.py [hunk 2]
```diff
@@ -1092,20 +1092,32 @@ async def upsert_suggestion(
...
+
     return result.scalar_one_or_none()
 
 
-async def delete_suggestion(db: "AsyncSession", suggestion: Suggestion) -> Suggestion:
-    return await suggestion.delete(db)
+async def delete_suggestion(db: "AsyncSession", search_engine: SearchEngine, suggestion: Suggestion) -> Suggestion:
+    async with db.begin_nested():
+        suggestion = await suggestion.delete(db, autocommit=False)
+        # TODO: Should we touch here dataset last_activity?
```
Comment (frascuchon) [inline src/argilla/server/contexts/datasets.py]: `last_activity` updates are related to user responses, so we can skip it for suggestions for now
```suggestion
```

**Review by frascuchon — APPROVED** [review]

**Review by gabrielmbmb — APPROVED** [review]

File: src/argilla/server/contexts/datasets.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/contexts/datasets.py]: It's clearly affected yes. I will take a look.

Comment (jfcalvo) [reply src/argilla/server/contexts/datasets.py]: We discussed it and we will take a first approach reindexing suggestions individually.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**